packages/zwave-js/src/lib/commandclass/ThermostatFanModeCC.ts

```diff
@@ -0,0 +1,390 @@
+				message: "querying supported thermostat fan modes...",
+				direction: "outbound",
+			});
+
+			const supportedModes = await api.getSupportedModes();
+			if (supportedModes) {
+				const logMessage = `received supported thermostat modes:${supportedModes
+					.map(
+						(mode) =>
+							`\n· ${getEnumMemberName(ThermostatFanMode, mode)}`,
+					)
+					.join("")}`;
+				this.driver.controllerLog.logNode(node.id, {
+					endpoint: this.endpointIndex,
+					message: logMessage,
+					direction: "inbound",
+				});
+			} else {
+				this.driver.controllerLog.logNode(node.id, {
+					endpoint: this.endpointIndex,
+					message:
+						"Querying supported thermostat fan modes timed out, skipping interview...",
+				});
+				return;
+			}
+		}
+
+		// Always query the actual status
+		this.driver.controllerLog.logNode(node.id, {
+			endpoint: this.endpointIndex,
+			message: "querying current thermostat fan mode...",
+			direction: "outbound",
+		});
+		const currentStatus = await api.get();
+		if (currentStatus) {
+			this.driver.controllerLog.logNode(node.id, {
+				endpoint: this.endpointIndex,
+				message:
+					"received current thermostat fan mode: " +
+					getEnumMemberName(ThermostatFanMode, currentStatus.mode),
+				direction: "inbound",
+			});
+		}
+
+		// Remember that the interview is complete
+		this.interviewComplete = true;
+	}
+}
+
+type ThermostatFanModeCCSetOptions = CCCommandOptions & {
+	mode: ThermostatFanMode;
+};
+
+@CCCommand(ThermostatFanModeCommand.Set)
+export class ThermostatFanModeCCSet extends ThermostatFanModeCC {
+	public constructor(
+		driver: Driver,
+		options:
+			| CommandClassDeserializationOptions
+			| ThermostatFanModeCCSetOptions,
+	) {
+		super(driver, options);
+		if (gotDeserializationOptions(options)) {
+			// TODO: Deserialize payload
+			throw new ZWaveError(
+				`${this.constructor.name}: deserialization not implemented`,
+				ZWaveErrorCodes.Deserialization_NotImplemented,
+			);
+		} else {
+			this.mode = options.mode;
+		}
+	}
+
+	public mode: ThermostatFanMode;
+
+	public serialize(): Buffer {
+		this.payload = Buffer.from([
+			// 0 R R R M M M M
+			this.mode & 0b1111,
+		]);
+		return super.serialize();
+	}
+
+	public toLogEntry(): MessageOrCCLogEntry {
+		const message: MessageRecord = {
+			mode: getEnumMemberName(ThermostatFanMode, this.mode),
+		};
+		return {
+			...super.toLogEntry(),
+			message,
+		};
+	}
+}
+
+@CCCommand(ThermostatFanModeCommand.Report)
+export class ThermostatFanModeCCReport extends ThermostatFanModeCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions,
+	) {
+		super(driver, options);
+
+		validatePayload(this.payload.length >= 1);
+		this._mode = this.payload[0] & 0b1111;
+
+		if (this.version >= 2) {
+			this._off = !!(this.payload[0] & 0b1000_0000);
+		}
+
+		this.persistValues();
+	}
+
+	private _mode: ThermostatFanMode;
+	@ccValue()
+	@ccValueMetadata({
+		...ValueMetadata.UInt8,
+		min: 0,
+		max: 15,
```

Comment:
This should be replaced with `states: enumValuesToMetadataStates(ThermostatFanMode)`, by default don't limit it to any modes. You're doing that in the `SupportedReport`.

Reply:
I essentially copied the metadata info from the ThermostatModeCC.ts file (which has min 0 and max 31), so I thought it was a guard on the size of the bitmask.  Might want to look at that CC too.

Reply:
Good point!